opencart 1.5.6.4/upload/catalog/controller/smailyforopencart/cron_customers.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Samaly subscribers synchronization.
+ */
+class ControllerSmailyForOpencartCronCustomers extends Controller {
+
+    public function index() {
+        // Customer model.
+        $this->load->model('account/customer');
+        // Load Smaily settings.
+        $this->load->model('setting/setting');
+        // Load Smaily helper.
+        $this->load->model('smailyforopencart/helper');
+        
+        $settings = $this->model_setting_setting->getSetting('smaily')['smaily_customer_sync'];
+
+        // Validate cron token.
+        if (!array_key_exists('token', $settings) ||
+            empty($this->request->get['token']) ||
+            $settings['token'] !== $this->request->get['token']
+        ) {
+            die('Unauthorized');
+        }
+        if ($settings['enabled'] == 1) {
+
+            // Get unsubscribers from smaily.
+            $unsubscribers_filter = array(
+                'list' => 2,
+            );
+
+            $unsubscribers = $this->model_smailyforopencart_helper->apiCall('contact', $unsubscribers_filter);
+            // Collect unsubscriber emails.
+            $unsubscribers_emails = [];
+            foreach ($unsubscribers as $unsubscriber) {
+                array_push($unsubscribers_emails, $unsubscriber['email']);
+            }
+            // Remove subscribed status from unsubscribed customers.
+            $subscribers = $this->model_smailyforopencart_helper->getSubscribedCustomers();
+            foreach ($subscribers as $subscriber) {
+                if (in_array($subscriber['email'], $unsubscribers_emails)) {
+                    $this->model_smailyforopencart_helper->unsubscribeCustomer($subscriber['customer_id']);
+                }
+            }
+            
+            // Get all users with subscribed status.
+            $subscribers = $this->model_smailyforopencart_helper->getSubscribedCustomers();
```

Comment:
Use similar pattern to subscribe contacts as with unsubscribing customers. But the difference is that you'd fetch customers from Opencart database in chunks, and just push the fetched chunk to Smaily.

Reply:
Would a LIMIT that loops higher on getSubscribedCustomers()'s mySQL query be a good idea to fetch customers in chunks or is there a better one?

Reply:
There is at least one other pattern besides SQL LIMIT, but using LIMIT is totally fine in this case.